### ISSUE TITLE
Hardcoded detection of available vision models means many vision models won't work

### ISSUE THREAD

**Issue by ProjectMoon** [Jeffser/Alpaca#313]
**Describe the bug**
Ability to use image recognition is controlled by checking `available_models.json`. This works until you have a model not in that file. Like https://ollama.com/library/minicpm-v.

**Expected behavior**
Any vision-capable model should be capable of using vision.

**Suggested Fix**
There is an ollama API called `/api/show`, which returns model info. Vision models have a section called `projector_info`, which contains the information for the image recognition part of the model. So if that key is present and not null, it can be assumed the model in question is a vision model.

_Note_: I'm not sure if ALL vision models have `projector_info` in the output, but every one I've used does (LLava + derivatives, MiniCPM). I'm guessing they all do, but I can't be certain. So maybe it's good to keep `available_models.json` check as a fallback.

**Debugging information**
Version 2.0.3.

**Comment by ProjectMoon** [comment]
I might try opening a PR for this today/tomorrow.

**Comment by atirut-w** [comment]
Note that you can still add by name, but good points.

**Comment by ProjectMoon** [comment]
Change is merged into main branch. I will test out the new version of the code when I'm able, to see if it behaves well with a remote instance of ollama with lots of models.

**Comment by ProjectMoon** [comment]
So startup takes maybe 20 seconds. Every time. It's iterating through all the models, calling `/show`.

**Comment by Jeffser** [comment]
/show only works with downloaded models, the `image recognition` tag is hardcoded to the list of available models, Moon contributed [this pr](https://github.com/Jeffser/Alpaca/pull/316) that makes it possible to use vision with any compatible models despite of what the hardcoded list says, for now I think this is the best we can do. I'll prepare a new release for later today

**Comment by ProjectMoon** [comment]
I know about the PR since I was the one who wrote it lol. My main test case is my remote ollama instance, which is exposed to the internet via Open WebUI.

**Comment by ProjectMoon** [comment]
Maybe parallel show requests on startup still a good idea?

**Comment by Jeffser** [comment]
The problem with having multiple show requests at the same time is that it _might_ crash Ollama

**Comment by ProjectMoon** [comment]
Well not if you limit them to like 5 at a time, I would imagine. The change as it is at the moment does technically work. But if you have several dozen models like me, it takes a while for the UI to be actually interacted with. And also if you try to quit Alpaca while it's loading, it'll crash (minor detail).

If parallel show requests on startup aren't possible, I think the best way would really be to call show ONCE when attempting to upload an image to the message. Then cache that result in memory for the model.

**Comment by Jeffser** [comment]
Alright I changed the way it works

1. Starts a different thread for each local model
2. Each model asks if they have vision when they are created in the UI
3. At the end the model selector is enabled

When new models are created / downloaded, they also ask if they have vision.

They don't call `/api/show` ever again, I made a variable that keeps track of their vision on each model

https://github.com/Jeffser/Alpaca/commit/df568c72171ffb055e7080fba57d692e07f55817
